dataset_and_model.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+"""
+The dataset object used in the multi-partner learning and contributivity measurement experiments.
+"""
+
+from sklearn.model_selection import train_test_split
+
+
+class DatasetAndModel:
```

Comment:
I would rename it to `Dataset `since it's mostly dataset related (there is only one function which relates to a model). And when you instantiate it, you name it `dataset_mnist `or `dataset_cifar`.

Reply:
Ok